### ISSUE TITLE
Proposed changes to CMake needed for AMReX

### ISSUE THREAD

**Issue by lucbv** [Exawind/nalu-wind#441]
@sayerhs @jrood-nrel @jhux2 @srajama1 
PR #436 introduces "modernization" in Nalu's CMake build requirements, unfortunately this was not advertised among the stakeholder of ExaWind, for instance `cmake_minimum_required (VERSION 3.13.0 FATAL_ERROR)` is problematic as Sandia is not currently providing modules for `CMake > sems-cmake/3.12.2`.

Also annoyingly these changes have broken my build on Summit so I can't work on the project anymore... I would like for these changes to be reverted as soon as possible to allow developers to work.

**Comment by jrood-nrel** [comment]
@lucbv I've temporarily reverted the changes I made to CMake. I apologize for any inconvenience it caused. Would you be able to request Sandia updates to newer versions of CMake on the machines you use so we can merge these changes? On Summit, CMake 3.15.2 exists, or exawind-builder can be updated to use a newer CMake if necessary there. A problem in the future will be that AMReX currently requires us to use at least CMake 3.14 for the AMR-Wind project and at some point that project will be strongly coupled to Nalu-Wind.

**Comment by jrood-nrel** [comment]
@lucbv Would you be able to do me a favor and try these changes but set the required CMake to 3.12.0? I may have incorrectly set the required version, but I don't have a CMake < 3.13.0 easily available to test.

**Comment by jrood-nrel** [comment]
@lucbv So I was actually able to test CMake 3.12.2 and I correctly listed CMake 3.13.0 as the minimum requirement for my updates. Does the latest master branch work for you again?

**Comment by lucbv** [comment]
Hi @jrood-nrel I am reopening this issue so we can maybe talk about it a bit.
I will make a request for a more recent CMake to be installed here but this will probably take a few days depending on our framework team workload.
Also I can send you the configuration script I am using on Summit, so you could see what goes wrong with them. I am setting the `--remove-duplicate-link` flag in Trilinos otherwise Nalu is not linking correctly. This apparently does not work with the changes you made.

**Comment by lucbv** [comment]
I am rebuilding on Summit right now and I will do a local build as well. I'll keep you updated, @jhux2 is using similar modules to mine I think so he might also be impacted by the higher CMake requirement.

**Comment by sayerhs** [comment]
@lucbv The original changes should also work with CMake 3.12. The requirement for 3.13 was unnecessary as @jrood-nrel mentions. So if you can test the original commit with just one change replace minimum version from 3.13 to 3.12 and test that on Sandia/Summit. 

As for AMReX and a higher CMake version, it is not a decision that we need to make at this time. I'd say we can do that closer to FY20-Q2.

**Comment by lucbv** [comment]
@sayerhs my build on Summit is using `-- CMAKE_VERSION='3.13.4'` so that would not be the issue, as reported above I think one of the flag was not accepted anymore yet it is needed for the build to complete...

**Comment by jrood-nrel** [comment]
@sayerhs Even if you use 3.13.x and set the minimum CMake version to 3.12.0, the changes should fail to compile from what I just tested.

@lucbv I will try the GPU build with my changes. I did not anticipate flags like `--remove-duplicate-link-files` in Trilinos to break with this.

**Comment by sayerhs** [comment]
@lucbv Ah, I see. thanks for the heads up.

**Comment by jrood-nrel** [comment]
I can confirm that `--remove-duplicate-link-files` is getting truncated to `-remove-duplicate-link-files` due to my changes. Sorry I did not catch this earlier. I will fix this problem with the code.

**Comment by lucbv** [comment]
@jrood-nrel thanks for looking into this and no worries about the issue with flags.
I would just recommend that when making significant changes in the CMake files of Nalu-Wind it would be best if these are shared among developers before PRs are merged. That way developers can report on how the changes are affecting currently work-flows and find work-around issues preemptively.

**Comment by jrood-nrel** [comment]
I fixed the issue with `--remove-duplicate-link-files`. I made another pull request as a draft with everyone as a reviewer that we can merge when we're more prepared for it. I probably shouldn't have brought AMReX into this since it is not immediately a requirement that we sync tools such as CMake between Exawind subprojects. It is just a minor justification for why I've been updating CMake in tandem for AMR-Wind and Nalu-Wind.
